staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -498,6 +482,13 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 				return nil, nil, apierrors.NewNotFound(qualifiedResource, name)
 			}
 			creating = true
+
+			// Given the existing object, get the new object
```

Comment:
Alternately, we could move the:
```
		if version == 0 {
			if !e.UpdateStrategy.AllowCreateOnUpdate() && !forceAllowCreate {
				return nil, nil, apierrors.NewNotFound(qualifiedResource, name)
			}
```
condition before running `UpdatedObject()`, what's best?

Reply:
Yeah, i was undecided as well but preferred this to having two version==0 at that time.

Reply:
I would tend to do:

1. `s/version/existingResourceVersion/`
2. `s/resourceVersion/newResourceVersion/`
3. Hoist this to the top of the modifier function
```go
err = e.Storage.GuaranteedUpdate(ctx, key, out, true, storagePreconditions, func(existing runtime.Object, res storage.ResponseMeta) (runtime.Object, *uint64, error) {
	existingResourceVersion, err := e.Storage.Versioner().ObjectResourceVersion(existing)
	if err != nil {
		return nil, nil, err
	}
	if existingResourceVersion == 0 {
		if !e.UpdateStrategy.AllowCreateOnUpdate() && !forceAllowCreate {
			return nil, nil, apierrors.NewNotFound(qualifiedResource, name)
		}
	}
	...
```

Reply:
Done